ffi/src/handle.rs

```diff
@@ -100,9 +113,19 @@ impl DatabaseHandle {
     ///
     /// If the path is empty, or if the configuration is invalid, this will return an error.
     pub fn new(args: DatabaseHandleArgs<'_>) -> Result<Self, api::Error> {
+        let root_store_path = args
+            .root_store_path
+            .as_str()
+            .map_err(|e| invalid_data(format!("root store path contains invalid utf-8: {e}")))?;
+
+        let root_store_dir = Some(root_store_path)
+            .filter(|s| !s.is_empty())
+            .map(PathBuf::from);
+
         let cfg = DbConfig::builder()
             .truncate(args.truncate)
             .manager(args.as_rev_manager_config()?)
+            .root_store_dir(root_store_dir)
```

Comment:
this works, but I think we need to switch to a directory per our discussion (probably a different PR).